tfhe/src/integer/parameters/mod.rs

```diff
@@ -117,3 +122,62 @@ pub const PARAM_MESSAGE_1_CARRY_1_KS_PBS_32_BITS: WopbsParameters = WopbsParamet
     ciphertext_modulus: CiphertextModulus::new_native(),
     encryption_key_choice: EncryptionKeyChoice::Big,
 };
+
+pub struct RadixCiphertextConformanceParams {
+    pub shortint_params: CiphertextConformanceParams,
+    pub num_blocks_per_integer: usize,
+}
+
+/// Structure to store the expected properties of a ciphertext
+/// Can be used on a server to check if client inputs are well formed
+/// before running a computation on them
+impl RadixCiphertextConformanceParams {
+    pub fn to_ct_list_conformance_parameters(
+        &self,
+        list_constraint: ListSizeConstraint,
+    ) -> RadixCompactCiphertextListConformanceParams {
+        RadixCompactCiphertextListConformanceParams {
+            shortint_params: self.shortint_params,
+            num_blocks_per_integer: self.num_blocks_per_integer,
+            num_integers_constraint: list_constraint,
+        }
+    }
+
+    pub fn from_classic_pbs_parameters(
+        classic_pbs_parameters: &ClassicPBSParameters,
+        num_blocks_per_integer: usize,
+    ) -> RadixCiphertextConformanceParams {
+        RadixCiphertextConformanceParams {
+            shortint_params: classic_pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer,
+        }
+    }
+
+    pub fn from_pbs_parameters(
+        pbs_parameters: &PBSParameters,
+        num_blocks_per_integer: usize,
+    ) -> RadixCiphertextConformanceParams {
+        RadixCiphertextConformanceParams {
+            shortint_params: pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer,
+        }
+    }
+}
```

Comment:
this could be only

`pub fn from_pbs_parameters<P: Into<PBSParameters>>(params: P, ...) { let params: PBSParameters = params.into();...}`

normally as PBSParameters is the "base enum" for ClassicPBSParameters and MultiBitPBS etc.

Reply:
Done